### PR TITLE
FOGL-2366: Migrate end_to_end_OCS test from bash to pytest 

### PR DESCRIPTION
**Pull request by Singhal-Vaibhav**
To test:
```
FogLAMP/tests/system/python $ pytest test_e2e_coap_OCS.py --ocs-client-id="<ocs-client-id>" --ocs-client-secret="ocs-client-secret" --ocs-tenant="ocs-tenant-id" --ocs-namespace=<ocs-namespace> --ocs-token="ocs-token"
```
Console output:
```
==== test session starts =====
collected 1 item                                                                                                                                                                       

Test system/python/test_e2e_coap_OCS.py Test that data is inserted in FogLAMP and sent to OCS
        start_south_north: Fixture that starts FogLAMP with south and north instance
        read_data_from_ocs: Fixture to read data from OCS
        Assertions:
            on endpoint GET /foglamp/asset
            on endpoint GET /foglamp/asset/<asset_name>
            data received from OCS is same as data sent .                                                                                                                        [100%]

==== 1 passed in 69.49 seconds ====
```

This pull will be rebased when FOGL-2322 is in

### CODE REVIEW

**Comment by ashish-jabble** [comment]
@Singhal-Vaibhav Please resolve conflicts!

File: tests/system/python/conftest.py [hunk 1]
```diff
@@ -15,6 +15,7 @@
 import base64
 import ssl
 import shutil
+import requests
```
Comment (praveen-garg) [inline tests/system/python/conftest.py]: Add a TODO, we should remove it soon.

File: tests/system/python/conftest.py [hunk 2]
```diff
@@ -126,6 +116,37 @@ def _start_north_pi_server_c(foglamp_url, pi_host, pi_port, north_plugin, pi_tok
 start_north_pi_server_c = start_north_pi_v2
 
 
+@pytest.fixture
+def start_north_ocs_v2():
```
Comment (praveen-garg) [inline tests/system/python/conftest.py]: Please remove from conftest and move to ocs test  or ocs fixtures

File: tests/system/python/conftest.py [hunk 3]
```diff
@@ -200,16 +221,69 @@ def _read_data_from_pi(host, admin, password, pi_database, asset, sensor):
     return _read_data_from_pi
 
 
+@pytest.fixture
+def read_data_from_ocs():
```
Comment (praveen-garg) [inline tests/system/python/conftest.py]: Please remove from conftest and move to ocs test or ocs fixtures

File: tests/system/python/test_e2e_coap_OCS.py [hunk 4]
```diff
@@ -0,0 +1,105 @@
+# -*- coding: utf-8 -*-
```
Comment (praveen-garg) [inline tests/system/python/test_e2e_coap_OCS.py]: name it `test_e2e_OCS.py` we should be able to pass south plugin name from supported list in longer term

File: tests/system/python/test_e2e_coap_OCS.py [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+        start_north_ocs_server_c: Fixture that starts OCS north task
+        remove_data_file: Fixture that remove data file created during the tests
+        remove_directories: Fixture that remove directories created during the tests"""
+
+    # Define the template file for fogbench
+    fogbench_template_path = os.path.join(
+        os.path.expandvars('${FOGLAMP_ROOT}'), 'data/{}'.format(TEMPLATE_NAME))
+    f = open(fogbench_template_path, "w")
+    f.write(
+        '[{"name": "%s", "sensor_values": '
```
Comment (praveen-garg) [inline tests/system/python/test_e2e_coap_OCS.py]: sensor_values is deprecated, we use readings

File: tests/system/python/test_e2e_coap_OCS.py [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+        '[{"name": "sensor", "type": "number", "min": %d, "max": %d, "precision": 0}]}]' % (
+            asset_name, SENSOR_VALUE, SENSOR_VALUE))
```
Comment (praveen-garg) [inline tests/system/python/test_e2e_coap_OCS.py]: sensor_values is deprecated, we use readings

File: tests/system/python/test_e2e_kafka.py [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+        assert ASSET_NAME == val[0]["assetCode"]
+        assert 1 == val[0]["count"]
+
+        conn.request("GET", '/foglamp/asset/{}'.format(ASSET_NAME))
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        val = json.loads(r)
+        assert 1 == len(val)
+        assert {'sensor': SENSOR_VALUE} == val[0]["reading"]
```
Comment (praveen-garg) [inline tests/system/python/test_e2e_kafka.py]: can sensor value be a nested dict or array?

File: tests/system/python/test_e2e_coap_OCS.py [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+SENSOR_VALUE = 20
+
+
+@pytest.fixture
+def prepare_template_reading_from_fogbench():
+    def _prepare_template_reading_from_fogbench(FOGBENCH_TEMPLATE, ASSET_NAME):
+        """ Define the template file for fogbench readings """
+        fogbench_template_path = os.path.join(
+            os.path.expandvars('${FOGLAMP_ROOT}'), 'data/{}'.format(FOGBENCH_TEMPLATE))
+        f = open(fogbench_template_path, "w")
```
Comment (praveen-garg) [inline tests/system/python/test_e2e_coap_OCS.py]: we should use `with` context

**Pull request merged**